tools/src/test/java/org/apache/kafka/tools/consumer/group/ShareGroupCommandTest.java

```diff
@@ -612,6 +615,189 @@ public void testGroupStatesFromString() {
+                Entry<String, String> res = ToolsTestUtils.grabConsoleOutputAndError(deleteOffsets(service));
+                String[] lines = res.getKey().trim().split("\n");
+                if (lines.length != 3 && !res.getValue().isEmpty()) {
+                    return false;
+                }
+
+                List<String> expectedResultHeader = List.of("TOPIC", "STATUS");
+                List<String> expectedResultValues1 = List.of(firstTopic, "Successful");
+                List<String> expectedResultValues2 = List.of(secondTopic, "Successful");
+
+                return Arrays.stream(lines[0].trim().split("\\s+")).toList().equals(expectedResultHeader) &&
+                    Arrays.stream(lines[1].trim().split("\\s+")).toList().equals(expectedResultValues1) &&
+                    Arrays.stream(lines[2].trim().split("\\s+")).toList().equals(expectedResultValues2);
+            }, "Expected a data row and no error in delete offsets result with group: " + firstGroup + " and topic: " + firstTopic);
+        }
+    }
+
+    @Test
+    public void testDeleteShareGroupOffsetsTopLevelError() throws Exception {
+        String firstGroup = "first-group";
+        String firstTopic = "t1";
+        String secondTopic = "t2";
+        String bootstrapServer = "localhost:9092";
+
+        List<String> cgcArgs = new ArrayList<>(Arrays.asList("--bootstrap-server", bootstrapServer, "--delete-offsets", "--group", firstGroup, "--topic", firstTopic, "--topic", secondTopic));
+        Admin adminClient = mock(KafkaAdminClient.class);
+        DeleteShareGroupOffsetsResult result = mock(DeleteShareGroupOffsetsResult.class);
+
+        KafkaFutureImpl<Void> resultFuture = new KafkaFutureImpl<>();
+        String errorMessage = "Group g3 not found.";
+        GroupIdNotFoundException exception = new GroupIdNotFoundException(errorMessage);
+
+        resultFuture.completeExceptionally(exception);
+        when(result.all()).thenReturn(resultFuture);
+
+        when(result.topicResult(eq(firstTopic))).thenReturn(resultFuture);
+        when(result.topicResult(eq(secondTopic))).thenReturn(resultFuture);
+
+        when(adminClient.deleteShareGroupOffsets(any(), any(), any())).thenReturn(result);
+
+        try (ShareGroupService service = getShareGroupService(cgcArgs.toArray(new String[0]), adminClient)) {
+            TestUtils.waitForCondition(() -> {
+                Entry<String, String> res = ToolsTestUtils.grabConsoleOutputAndError(deleteOffsets(service));
+                String[] lines = res.getKey().trim().split("\n");
+                if (lines.length != 5 && !res.getValue().isEmpty()) {
+                    return false;
+                }
+
+                List<String> errorLine = Stream.concat(
+                    Stream.of("Error:"),
+                    Arrays.stream(errorMessage.trim().split("\\s+"))
+                    ).toList();
+
+                List<String> firstLine = new ArrayList<>(errorLine);
```

Comment:
This could be a bit cleaner if you gave the lines more descriptive names, such as `errorLine`, `expectedResultHeader`, `expectedResultValue1` and `expectedResultValue2`. Then the test cases would be more consistent with each other.